### PR TITLE
 Can swtich to use lager as backend to support throttling

### PR DESCRIPTION
**Pull request by windkit**

### CODE REVIEW

File: src/lager_logger_api.erl [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    lager_app:configure_sink(LogGroup,
+                             [{handlers, [{lager_file_backend, [{file, LogFileName}, {level, info},
+                                                                {size, 10485760}, {date, "$D0"},
+                                                                {formatter, lager_default_formatter},
+                                                                {formatter_config, [message, "\n"]},
+                                                                {rotator, leo_logger_rotator}
+                                                               ]}]},
+                              {async_threshold, 500},
+                              {async_threshold_window, 50}]),
+    ets:insert(?LOG_ID_TO_SINK_ETS, {LogId, {LogGroup, LogFileName}}),
```
Comment (mocchira) [inline src/lager_logger_api.erl]: those parameters also should be chosen based on the result of benchmark I suggested on the above comment. also let me confirm async_threshold(_window) are absent on the setting of the leveled error logger above. is it intentional?

**Comment by mocchira** [comment]
@windkit one thing I missed to point out on the review.
Please add the rotator behavior into src/leo_logger_rotator.erl

File: src/lager_logger_api.erl [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    new(RootPath, Level, []).
+
+-spec(new(RootPath, Level, Loggers) ->
+             ok when RootPath::string(),
+                     Level::integer(),
+                     Loggers::[{atom(), log_appender()}]).
+new(RootPath, Level, _Loggers) ->
+    application:set_env(lager, log_root, RootPath),
+    application:set_env(lager, crash_log, "crash.log"),
+    application:set_env(lager, error_logger_hwm, 500),
```
[the remark this answers is not in the document]
Comment (windkit) [reply src/lager_logger_api.erl]: @mocchira with one storage node stopped, there is not much log entries output.
1 Line in 5 seconds. 
```
[E]     S2@192.168.100.38       2017-10-16 08:29:19.201660 +0900        1508110159      null:null       0       {module,"leo_membership_cluster_local"},{function,"compare_with_remote_chksum/3"},{line,405},{body,{'S1@192.168.100.37',nodedown}}
```

Instead, I stooped one storage node `S1@192.168.100.37`, remove the avs directory `/ssd/avs` and restart it. So background recovery is in place.
Number of logs per second is around 2000, and the frontend performance is almost the same.
The `error_logger_hwm` limits the number of message log through `error_logger`.

I will set the `killer_hwm` instead, so the logger will be restarted when number of pending messages exceed the limit, what do you think?

[the remark this answers is not in the document]
Comment (mocchira) [reply src/lager_logger_api.erl]: @windkit Confirmed each spec and killer_hwm looks more safe bet than error_logger_hwm especially when bursting logs keep happening for a long time so agreed with your suggestion.

**Pull request merged**